modalsheet/src/main/java/eu/wewox/modalsheet/ModalSheet.kt

```diff
@@ -95,260 +84,158 @@ public fun <T> ModalSheet(
+    content: @Composable ColumnScope.() -> Unit,
 ) {
-    // If content animated by AnimatedVisibility in PopupBody is still visible (in composition)
-    var contentVisible by remember { mutableStateOf(false) }
+    val sheetState = rememberModalBottomSheetState(
+        skipHalfExpanded = true,
+        initialValue = ModalBottomSheetValue.Hidden,
+        confirmStateChange = {
+            // Intercept and disallow hide gesture / action
+            if (it == ModalBottomSheetValue.Hidden && !cancelable) {
+                return@rememberModalBottomSheetState false
+            }
+
+            onVisibleChange(it == ModalBottomSheetValue.Expanded)
+
+            true
+        }
+    )
 
-    if (visible || contentVisible) {
-        FullscreenPopup(
-            onDismiss = onDismiss
-        ) {
-            PopupBody(
-                visible = visible,
-                scrim = scrim,
-                content = {
-                    // Set contentVisible true when the composable enters composition and false when it leaves
-                    DisposableEffect(Unit) {
-                        contentVisible = true
-                        onDispose { contentVisible = false }
-                    }
+    if (!visible && sheetState.progress.to == sheetState.progress.from && !sheetState.isVisible) {
+        return
+    }
 
-                    val scope = remember { ModalSheetScopeImpl() }
-                    SwipeableContent(
-                        swipeEnabled = swipeEnabled,
-                        onSwipedAway = onDismiss,
-                        scrollState = scope.scrollState
-                    ) {
-                        Surface(
-                            shape = shape,
-                            color = backgroundColor,
-                            modifier = Modifier.fillMaxWidth()
-                        ) {
-                            scope.content()
-                        }
-                    }
-                }
-            )
+    LaunchedEffect(visible) {
+        if (visible) {
+            sheetState.show()
+        } else {
+            sheetState.hide()
         }
     }
-}
-
-/**
- * The scope for [ModalSheet] content.
- */
-@ExperimentalSheetApi
-public interface ModalSheetScope {
 
-    /**
-     * The scroll state.
-     * Must be passed to [.verticalScroll] modifier to support scrolling inside content.
-     */
-    public val scrollState: ScrollState
+    ModalSheet(
+        sheetState = sheetState,
+        onDismiss = {
+            if (cancelable) {
+                onVisibleChange(false)
+            }
+        },
+        shape = shape,
+        elevation = elevation,
+        backgroundColor = backgroundColor,
+        contentColor = contentColor,
+        scrimColor = scrimColor,
+        content = content,
+    )
 }
 
 /**
- * Contains the default values used by [ModalSheet].
+ * Static modal sheet that behaves like bottom sheet and draws over system UI.
```

Comment:
What do you mean by "static"?

Reply:
I paraphrase it a little bit  😇

Reply:
It still says "Static modal sheet..." without a change. Maybe you forgot to change this one?

Reply:
Ooops, I have paraphrased another one 🤦 

I just removed "static" here as this one could be quite custom and "static" was pretty misleading.